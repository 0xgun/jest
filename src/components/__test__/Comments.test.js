import { render, waitFor, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Comments from "../Comments";
import axios from "axios";
jest.mock("axios");
const dummyComments = [
  {
    postId: 1,
    id: 1,
    name: "id labore ex et quam laborum",
    email: "Eliseo@gardner.biz",
    body: "laudantium enim quasi est quidem magnam voluptate ipsam eos\ntempora quo necessitatibus\ndolor quam autem quasi\nreiciendis et nam sapiente accusantium",
  },
  {
    postId: 1,
    id: 2,
    name: "quo vero reiciendis velit similique earum",
    email: "Jayne_Kuhic@sydney.com",
    body: "est natus enim nihil est dolore omnis voluptatem numquam\net omnis occaecati quod ullam at\nvoluptatem error expedita pariatur\nnihil sint nostrum voluptatem reiciendis et",
  },
];

describe("Comments component", () => {
  test("should get all comments", async () => {
    axios.get.mockResolvedValue({
      //  .mockResolvedValue
      data: dummyComments,
    });
    render(<Comments />);
    const comments = await waitFor(() => screen.findAllByTestId("comments"));
    expect(comments).toHaveLength(2);
  });
});

describe("Comments Snapshot", () => {
  it("should matches DOM shapshot", () => {
    const tree = renderer.create(<Comments />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
