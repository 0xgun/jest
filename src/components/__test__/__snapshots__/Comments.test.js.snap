// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments Snapshot should matches DOM shapshot 1`] = `
<div>
  <p>
    Comments
  </p>
  <ul />
</div>
`;
